components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,254 @@
+  topicName: string
+  type: string
+}
+
+type Notifications = NotificationProps[]
+
+export default function Newsfeed() {
+  const isMobile = useMediaQuery("(max-width: 768px)")
+
+  const { user } = useAuth()
+  const uid = user?.uid
+  const { result: profile, loading } = usePublicProfile(uid)
+  const subscriptionRef = collection(
+    firestore,
+    `/users/${uid}/userNotificationFeed/`
+  )
+
+  const [isFilteringOrgs, setIsFilteringOrgs] = useState<boolean>(false)
+  const [isFilteringBills, setIsFilteringBills] = useState<boolean>(false)
+
+  const [allResults, setAllResults] = useState<Notifications>([])
+  const [orgResults, setOrgResults] = useState<Notifications>([])
+  const [billResults, setBillResults] = useState<Notifications>([])
+
+  const [notificationsDisplay, setNotificationsDisplay] =
+    useState<Notifications>([])
+  const [shouldDisplayResults, setShouldDisplayResults] =
+    useState<boolean>(true)
+
+  const onOrgFilterChange = (isFiltering: boolean) => {
+    const filterUpdatingTo = !isFiltering
+    setIsFilteringOrgs(filterUpdatingTo)
+
+    // isFilteringOrgs has to be inverted because setState (i.e. setIsFilteringOrgs)
+    // doesn't happen until a batch at the end
+
+    setShouldDisplayResults(true)
+    if (!isFilteringOrgs && isFilteringBills) {
+      return setShouldDisplayResults(false)
+    } else if (!isFilteringOrgs) {
+      return setNotificationsDisplay(billResults)
+    } else if (isFilteringBills) {
+      return setNotificationsDisplay(orgResults)
+    }
+    return setNotificationsDisplay(allResults)
+  }
+
+  const onBillFilterChange = (isFiltering: boolean) => {
+    const filterUpdatingTo = !isFiltering
+    setIsFilteringBills(filterUpdatingTo)
+
+    // isFilteringBills has to be inverted because setState (i.e. setIsFilteringBills)
+    // doesn't happen until a batch at the end
+
+    setShouldDisplayResults(true)
+    if (isFilteringOrgs && !isFilteringBills) {
+      return setShouldDisplayResults(false)
+    } else if (isFilteringOrgs) {
+      return setNotificationsDisplay(billResults)
+    } else if (!isFilteringBills) {
+      return setNotificationsDisplay(orgResults)
+    }
+    return setNotificationsDisplay(allResults)
+  }
+
+  const notificationQuery = async () => {
+    let notificationList: Notifications = []
+    const q = query(subscriptionRef, where("uid", "==", `${uid}`))
+    const querySnapshot = await getDocs(q)
+    querySnapshot.forEach(doc => {
+      // doc.data() is never undefined for query doc snapshots
+      notificationList.push(doc.data().notification)
+    })
+
+    if (notificationsDisplay.length === 0 && notificationList.length != 0) {
+      setAllResults(notificationList)
+      setOrgResults(
+        notificationList.filter(notification => notification.type === "org")
+      )
+      setBillResults(
+        notificationList.filter(notification => notification.type === "bill")
+      )
+      setNotificationsDisplay(allResults)
+    }
+  }
```

Comment:
Once lines 97-105 can be removed after removing redundant state, I think this function could be factored out into its own function, outside of the body of the component, since it deals solely with retrieving the notifications from a database, and has nothing specifically to do with React:

```suggestion
  async function notificationQuery(uid: string) => {
    let notificationList: Notifications = []
    const q = query(subscriptionRef, where("uid", "==", `${uid}`))
    const querySnapshot = await getDocs(q)
    querySnapshot.forEach(doc => {
      // doc.data() is never undefined for query doc snapshots
      notificationList.push(doc.data().notification)
    })
    return notificationList
  }
```

Then we can use it in the useEffect, where we care about setting state:

```js
useEffect(() => {
  notificationQuery(uid).then(notifications => setAllResults(notifications));
}, [uid, setAllResults]);
```

Following a [principle of separations of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) can help make code more re-usable, as well as more readable.

In this instance, we're separating out _how the data is retrieved_, e.g. via Firebase, from _how the data is displayed_, e.g. via React `useState` setter. It's not always this clean, but attempting to make it that clean is a worthy pursuit in my opinion :)